### PR TITLE
Log through pipes

### PR DESCRIPTION
**Pull request by andreibancioiu**
https://github.com/ElrondNetwork/elrond-go-logger/blob/pipes/pipes/README.md

### CODE REVIEW

File: pipes/parentPart.go [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+func (part *parentPart) forwardProfile() {
+	profile := logger.GetCurrentProfile()
+	part.messenger.SendProfile(profile)
+}
+
+func (part *parentPart) continuouslyRead(childStdout io.Reader, childStderr io.Reader) {
+	stdoutReader := bufio.NewReader(childStdout)
+	stderrReader := bufio.NewReader(childStderr)
+
+	go func() {
```
Comment (AdoAdoAdo) [inline pipes/parentPart.go:98]: can you define the functions to call on go routines outside of countinuouslyRead?

**Review by AdoAdoAdo — DISMISSED** [review]

File: pipes/childPart.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+	if err != nil {
+		return err
+	}
+
+	go part.continuouslyReadProfile()
+	return nil
+}
+
+func (part *childPart) addAsObserver() error {
+	part.outputSubject.ClearObservers()
```
Comment (iulianpascalau) [inline pipes/childPart.go:56]: this is somehow misleading as add will actually do a clear, removing all other observers

File: pipes/childPart.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+	}
+
+	return &childPart{
+		messenger:        NewChildMessenger(profileReader, logsWriter),
+		outputSubject:    logger.GetLogOutputSubject(),
+		logLineFormatter: logLineFormatter,
+	}, nil
+}
+
+func (part *childPart) StartLoop() error {
```
Comment (iulianpascalau) [inline pipes/childPart.go:39]: Missing comments on exported elements

File: pipes/childPart.go (hunk 2 above)
Comment (andreibancioiu) [reply pipes/childPart.go:56]: Renamed to `registerAsSoleObserver`.

File: pipes/childPart.go (hunk 3 above)
Comment (andreibancioiu) [reply pipes/childPart.go:40]: Added comments.

File: pipes/parentPart.go (hunk 1 above)
Comment (andreibancioiu) [reply pipes/parentPart.go:101]: Fixed, extracted functions.

**Review by AdoAdoAdo — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Pull request merged**